payments-core/src/main/java/com/stripe/android/IntentConfirmationInterceptor.kt

```diff
@@ -0,0 +1,226 @@
+        return if (clientSecret != null) {
+            createConfirmStep(
+                clientSecret = clientSecret,
+                shippingValues = shippingValues,
+                paymentMethodCreateParams = paymentMethodCreateParams,
+                setupForFutureUsage = setupForFutureUsage,
+            )
+        } else {
+            val paymentMethod = createPaymentMethod(paymentMethodCreateParams)
+
+            if (paymentMethod != null) {
+                intercept(
+                    clientSecret = null,
+                    paymentMethod = paymentMethod,
+                    shippingValues = shippingValues,
+                    setupForFutureUsage = setupForFutureUsage,
+                )
+            } else {
+                IntentConfirmationInterceptor.NextStep.Fail(
+                    error = IllegalStateException(
+                        "Failed to create payment method during confirmation"
+                    ),
+                )
+            }
+        }
+    }
+
+    override suspend fun intercept(
+        clientSecret: String?,
+        paymentMethod: PaymentMethod,
+        shippingValues: ConfirmPaymentIntentParams.Shipping?,
+        setupForFutureUsage: ConfirmPaymentIntentParams.SetupFutureUsage?,
+    ): IntentConfirmationInterceptor.NextStep {
+        return if (clientSecret != null) {
+            createConfirmStep(clientSecret, shippingValues, paymentMethod)
+        } else {
+            when (createIntentCallback) {
+                is CreateIntentCallbackForServerSideConfirmation -> {
+                    handleServerSideConfirmation(
+                        createIntentCallback = createIntentCallback,
+                        shouldSavePaymentMethod = setupForFutureUsage == OffSession,
+                        paymentMethod = paymentMethod,
+                        shippingValues = shippingValues
+                    )
+                }
+                is CreateIntentCallback -> {
+                    handleClientSideConfirmation(
+                        createIntentCallback = createIntentCallback,
+                        paymentMethod = paymentMethod,
+                        shippingValues = shippingValues
+                    )
+                }
+                else -> {
+                    IntentConfirmationInterceptor.NextStep.Fail(
+                        error = IllegalStateException(
+                            "ConfirmCallback must be implemented when using " +
+                                "IntentConfiguration to configure PaymentSheet",
+                        ),
+                    )
+                }
+            }
+        }
+    }
+
+    private suspend fun createPaymentMethod(params: PaymentMethodCreateParams): PaymentMethod? {
+        return runCatching {
+            stripeRepository.createPaymentMethod(
+                paymentMethodCreateParams = params,
+                options = requestOptions,
+            )
+        }.getOrNull()
```

Comment:
Are we essentially swallowing this error?

Reply:
Yes, that’s what’s happening right now. Though I guess we can pass it to the caller and populate `IntentConfirmationInterceptor.NextStep.Fail` with it. I’ll make that change.